epochStart/metachain/systemSCs.go

```diff
@@ -40,6 +40,9 @@ type systemSCProcessor struct {
 	validatorInfoCreator    epochStart.ValidatorInfoCreator
 	endOfEpochCallerAddress []byte
 	stakingSCAddress        []byte
+
+	mapNumSwitchedPerShard   map[uint32]uint32
```

Comment:
These maps are used without mutex protection. Please check if they should be used under the mutex or not.

Reply:
Not possible.

Reply:
It is used in serial way only